packages/roosterjs-editor-api/lib/utils/processList.ts

```diff
@@ -5,76 +5,78 @@ import {
     isVoidHtmlElement,
     isBlockElement,
     Browser,
-    isNodeEmpty,
-    getTagOfNode,
-    getComputedStyles,
-    applyFormat,
-    getFirstLeafNode,
+    getSelectionPath,
+    getRangeFromSelectionPath,
 } from 'roosterjs-editor-dom';
 
 const TEMP_NODE_CLASS = 'ROOSTERJS_TEMP_NODE_FOR_LIST';
-const TEMP_NODE_HTML = `<img class="${TEMP_NODE_CLASS}">`;
+const TEMP_NODE_HTML = "<img class=\"" + TEMP_NODE_CLASS + "\">";
+//const SELECTION_NODE_CLASS = 'ROOSTERJS_SELECTION_NODE';
+//const SELECTION_NODE_HTML = "<img class=\"" + SELECTION_NODE_CLASS + "\">";
+
+type ValidProcessListDocumentCommands = DocumentCommand.Outdent | DocumentCommand.Indent | DocumentCommand.InsertOrderedList | DocumentCommand.InsertUnorderedList;
 
 /**
  * Browsers don't handle bullet/numbering list well, especially the formats when switching list statue
  * So we workaround it by always adding format to list element
  */
-export default function processList(editor: Editor, command: DocumentCommand): Node {
-    if (Browser.isChrome) {
+export default function processList(editor: Editor, command: ValidProcessListDocumentCommands): Node {
+    let clonedNode: Node;
+    let relativeSelectionPath;
+    if (Browser.isChrome && command == DocumentCommand.Outdent) {
+        const parentLINode =  editor.getElementAtCursor('LI');
+        if (parentLINode) {
+            let currentRange = editor.getSelectionRange();
+            relativeSelectionPath = getSelectionPath(parentLINode, currentRange);

```

Comment:
Please do verify if currentRange is not collapsed and end position is not under parenLINode. e.g.
[LI]test{SelectionStart}test[/LI]
[LI]test{SelectionEnd}test[/LI]

Because in that case end position is out of the relative node (parentLiNode), not sure if it still work in that case

Reply:
Good point. Let me test a few scenarios with that.

Reply:
Yeah, multiple outdents are midly busted, but I don't think they're any more busted than before. So, I'll check to make sure the range is collapsed before continuing the process.